torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -291,16 +279,19 @@ class ProcessGroupNCCL : public ProcessGroup {
     // this callback. This new FutureNCCL's cudaEvents will record the
     // callback's stream and will have the result value of the callback.
     void addCallback(std::function<void(void)> callback) override {
+      // FIXME Should we find a way to allow to change the priority of streams?
```

Comment:
I commented in https://github.com/pytorch/pytorch/issues/48305#issuecomment-734458551 with a few ideas on how to tackle this.